src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -46,6 +46,9 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
                 logging.Services.AddSingleton<IWorkerDiagnostics, GrpcWorkerDiagnostics>();
             });
 
+            // FunctionMetadataProvider for worker driven function-indexing
+            services.AddSingleton<IFunctionMetadataProvider, DefaultFunctionMetadataProvider>();
```

Comment:
Should this be in `AddGrpc` method? this DI registration is not something needed to enable GRPC communication. rite? Should this go in `AddFunctionsWorkerCore` ?

Reply:
`IFunctionMetadataProvider` and `DefaultFunctionMetadataProvider` are in the Grpc project not the Core project. They are tied to the Grpc project because they're dependent on the proto file (deserializing the metadata file to grpc messages defined in the protofile). The proto-file is needed because we are using it to avoid a webjobs dependency (which the current GA function loader and sdk uses).

Moving this to Core and registering it in `AddFunctionsWorkerCore` would require some refactoring and creating a dependency on the gRPC project if we want to use the protofile. If we don't want to use the protofile or the webjobs dependency, and recreate our own functionmetadata and binding objects, we'd also have to maintain it to be consistent with the host / grpc definitions. 

This is what I remember from previous discussions / design questions I had from earlier iterations, but you are correct that this isn't something needed for grpc communication and would normally be part of the core project.